telegram/inputfile.py

```diff
@@ -67,7 +67,10 @@ def __init__(self,
 
         if isinstance(self.input_file, file):
             self.input_file_content = self.input_file.read()
-            self.filename = os.path.basename(self.input_file.name)
+            if 'filename' in self.data and self.data['filename']:
+                self.filename = self.data['filename']
```

Comment:
Instead ``self.filename = self.data.pop('filename')`` otherwise I'll pass along Telegram API and it will raise an error because ``filename`` is not a known parameter.